service/host.go

```diff
@@ -66,3 +66,11 @@ func (host *serviceHost) GetExtensions() map[component.ID]component.Component {
 func (host *serviceHost) GetExporters() map[component.DataType]map[component.ID]component.Component {
 	return host.pipelines.GetExporters()
 }
+
+func (host *serviceHost) notifyComponentStatusChange(source *component.InstanceID, event *component.StatusEvent) {
+	// TODO: What should we do if there is an error returned by a StatusWatcher?
```

Comment:
Good question. This could cause an explosion of errors if they are reported back to the status watcher. 

Is there anything useful we can do with an error from the watcher? Maybe we just log it in the watcher and never return an error?

Reply:
I removed the error return value. It turns out it was never being set anyways. A StatusWatcher should handle its own errors internally.